src/jit/lower.cpp

```diff
@@ -5297,10 +5301,13 @@ void Lowering::DoPhase()
 //
 void Lowering::CheckCallArg(GenTree* arg)
 {
-    if (arg->OperIsStore() || arg->IsArgPlaceHolderNode() || arg->IsNothingNode() || arg->OperIsCopyBlkOp())
+    if ((arg->OperIsStore() && !arg->IsValue()) || arg->IsArgPlaceHolderNode() || arg->IsNothingNode() ||
+        arg->OperIsCopyBlkOp())
```

Comment:
Likewise could we just check `arg->IsValue() || arg->OperIsPutArgStk() ` here instead?